testsuite/cluster-test/src/experiments/versioning_test.rs

```diff
@@ -144,12 +128,52 @@ impl Experiment for ValidatorVersioning {
             .map_err(|e| format_err!("Failed to create signed transaction: {}", e))
         };
 
-        let txn1 = txn_gen(&mut account_1)?;
+        // grab a validator node
+        let old_validator_node = context.cluster.random_validator_instance();
+        let mut old_client = old_validator_node.json_rpc_client();
 
-        execute_and_wait_transactions(&mut full_node_client, &mut account_1, vec![txn1.clone()])
-            .await?;
+        info!("1. Send a transaction using the new feature to a validator node");
+        let txn1 = txn_gen(&account)?;
+        if execute_and_wait_transactions(&mut old_client, &mut account, vec![txn1])
+            .await
+            .is_ok()
+        {
+            return Err(format_err!(
+                "The transaction should be rejected as the new feature is not yet recognized \
+                by any of the validator nodes"
+            ));
+        };
+        info!("-- [Expected] The transaction is rejected by the validator node");
+
+        info!("2. Update the first batch of validator nodes");
+        update_batch_instance(
+            context,
+            &self.first_batch,
+            &self.first_batch_lsr,
+            self.updated_image_tag.clone(),
+        )
+        .await?;
+
+        // choose an updated validator
+        let new_validator_node = self
+            .first_batch
+            .get(0)
+            .expect("getting an updated validator instance requires a non-empty list");
+        let mut new_client = new_validator_node.json_rpc_client();
+
+        info!("3. Send the transaction using the new feature to an updated validator node");
+        let txn3 = txn_gen(&account)?;
+        if execute_and_wait_transactions(&mut new_client, &mut account, vec![txn3])
+            .await
+            .is_ok()
+        {
+            return Err(format_err!(
+                "The transaction should be rejected as the feature is under gating",
+            ));
+        }
+        info!("-- The transaction is rejected as expected");
```

Comment:
Do we have any guarantee of network liveness during the execution step? If not maybe emit some transaction traffic to make sure the network is still functional all the time? cc @zihao-c

Reply:
what do you mean by network liveness? CT has mechanism to retry submit, and then it will keep querying seq number until tx is expired or tx is committed and seq number match

Reply:
base on that I do think network is live during execution